include/mc_tasks/ImpedanceTask.h

```diff
@@ -0,0 +1,359 @@
+  }
+
+  /*! \brief Set the desired velocity of the surface in the world frame. */
+  void desiredVel(const sva::MotionVecd & vel)
+  {
+    desiredVelW_ = vel;
+  }
+
+  /*! \brief Get the desired acceleration of the surface in the world frame. */
+  const sva::MotionVecd & desiredAccel() const noexcept
+  {
+    return desiredAccelW_;
+  }
+
+  /*! \brief Set the desired acceleration of the surface in the world frame. */
+  void desiredAccel(const sva::MotionVecd & accel)
+  {
+    desiredAccelW_ = accel;
+  }
+
+  /*! \brief Get the relative pose from desired frame to compliance frame represented in the world frame. */
+  const sva::PTransformd & deltaCompliancePose() const
+  {
+    return deltaCompPoseW_;
+  }
+
+  /*! \brief Get the compliance pose of the surface in the world frame.
+   *
+   *  \note Compliance pose cannot be set by user because it is calculated from the impedance equation internally.
+   *  See the Constructor description for the definition of compliance pose.
+   *
+   */
+  const sva::PTransformd compliancePose() const
+  {
+    sva::PTransformd T_0_d(desiredPoseW_.rotation());
+    return T_0_d * deltaCompPoseW_ * T_0_d.inv() * desiredPoseW_;
+  }
+
+  /*! \brief Get the target wrench in the surface frame. */
+  const sva::ForceVecd & targetWrench() const noexcept
+  {
+    return targetWrench_;
+  }
+
+  /*! \brief Set the target wrench in the world frame.
+   * This function will convert the wrench from the world frame to the surface frame, and call targetWrench().
+   *
+   */
+  void targetWrenchW(const sva::ForceVecd & wrenchW)
+  {
+    const auto & X_0_s = robots.robot(rIndex).surfacePose(surfaceName);
+    targetWrench(X_0_s.dualMul(wrenchW));
+  }
+
+  /*! \brief Set the target wrench in the surface frame. */
+  void targetWrench(const sva::ForceVecd & wrench)
+  {
+    targetWrench_ = wrench;
+  }
+
+  /*! \brief Get the measured wrench in the surface frame. */
+  const sva::ForceVecd & measuredWrench() const
+  {
+    return measuredWrench_;
+  }
+
+  /*! \brief Get the filtered measured wrench in the surface frame. */
+  const sva::ForceVecd & filteredMeasuredWrench() const
+  {
+    return filteredMeasuredWrench_;
+  }
+
+  /*! \brief Get the cutoff period for the low-pass filter of measured wrench. */
+  double cutoffPeriod()
+  {
+    return lowPass_.cutoffPeriod();
+  }
+
+  /*! \brief Set the cutoff period for the low-pass filter of measured wrench. */
+  void cutoffPeriod(double cutoffPeriod)
+  {
+    lowPass_.cutoffPeriod(cutoffPeriod);
+  }
+
+  /*! \brief Load parameters from a Configuration object. */
+  void load(mc_solver::QPSolver & solver, const mc_rtc::Configuration & config) override;
+
+protected:
+  // Impedance parameters represented in the surface frame
+  sva::ForceVecd impM_; // must be set in the Constructor
+  sva::ForceVecd impD_; // must be set in the Constructor
+  sva::ForceVecd impK_; // must be set in the Constructor
+  sva::MotionVecd wrenchGain_ = sva::MotionVecd::Zero();
```

Comment:
[`sva::ImpedanceVecd`](https://github.com/jrl-umi3218/SpaceVecAlg/blob/master/src/SpaceVecAlg/ImpedanceVec.h) is a better match for this kind of parameters, it's similar to `sva::ForceVecd` but has `linear()` and `angular()` accessors which are more explicit. Although they are not super well supported by mc_rtc so I will make a PR for this later today.

Reply:
> Although they are not super well supported by mc_rtc so I will make a PR for this later today.

This is done in #112